### DISCUSSION TITLE
Did I just get bamboozled by the words "open source" again?

### DISCUSSION THREAD

**Discussion by Hubro** [tinacms/tinacms#3972] [General]
My first impression of Tina was that it's much like Strapi, i.e. a self hosted headless CMS, but built on Git and Markdown files. As it was also open source, this sounded extremely interesting.

After spending a couple of hours going through the docs, I'm starting to feel tricked. Is Tina seriously branded as open source when the _entire point_ of the product (i.e. **the headless CMS**) is a proprietary hosted service?

From the docs I didn't find any reference on how to "go to production" without Tina Cloud. Please tell me I've misunderstood.

**Comment by mAAdhaTTah** [comment]
See here: https://github.com/tinacms/tinacms/discussions/3589

**Reply by Hubro** [reply]
Interesting, so it can be self-hosted. Still, the license states:

> Use Limitation: 5 users, 2 roles, and 500 documents per website

So it seems like Tina can't be used for any real websites without paying for a license or using the hosted service, so it would be utter cackling madness to call it _"open source"_. Yet, that's what WebToolsWeekly wrote, which is what brought me here:

![image](https://github.com/tinacms/tinacms/assets/597206/b5b0b83f-c67e-483a-bdf4-b37ef1b7d71e)

And what a cursory Google search will confirm:

![image](https://github.com/tinacms/tinacms/assets/597206/8c9e7c93-1184-4f2a-9fd8-a0d3b8fef456)

![image](https://github.com/tinacms/tinacms/assets/597206/fe306cf4-c18e-460f-845b-a675e4498592)

Oh, and their own website:

![image](https://github.com/tinacms/tinacms/assets/597206/0ca45ec7-18c8-40fc-a3db-110166a7c88e)

I'm utterly confused, is Tina open source or not?

---

EDIT: Found [the license FAQ](https://github.com/tinacms/tinacms/blob/main/packages/@tinacms/datalayer/LICENSE-FAQ.md), so it seems Tina is open source in name only. It claims to be an open source project, which is great for PR, but you're not actually allowed to use it for real websites without paying a commercial license or using their cloud hosted solution.

Dummies like me take them at their word, get hyped by the features and the fact that it's "open source" and invest time and energy into learning how to use it. Classic bait and switch. How is this anything other than slimy, deceptive marketing?

**Reply by SGiumelli** [reply]
Open source is not free software. It has never been. Isn't it a little strange to blame others for your own misconceptions ?

I understand your feelings and you are 100% right to ask for clarifications. But the accusatory wording won't help.

Strapi had limitations on self hosted licence. Limited number of role, etc.

The point of those limitations is to allow individual to use it for free and makes companies to pay.
The one thing here is the the threshold is low for documents even for individual.

**Reply by Hubro** [reply]
> Isn't it a little strange to blame others for your own misconceptions ?

Not when it's the common conception. I think 99/100 developers would be surprised *and angry* if you told them that they couldn't use an _open source_ project without paying for a license. For fun I asked ChatGPT what it thought, and it said:

> One of the key principles of open source is the concept of "free software," which emphasizes the freedom to use, study, modify, and distribute software. Open source software is typically licensed under licenses approved by the Open Source Initiative (OSI), such as the GNU General Public License (GPL) or the MIT License. These licenses ensure that the software remains open and accessible to the community, even as it evolves.

In my 12 years in the industry, this is the first time I've heard of anything being open source but *not free to use*, especially when it's being advertised as open source in media. It doesn't help that I read most of the documentation and was ready to start using this project before I even started to suspect this wasn't free to use. That makes this feel like an elaborate, dishonest bait and switch.

The least they could do is make it a lot more clear in their documentation that Tina is a paid product.

**Reply by scottgallant** [reply]
@Hubro, thanks for bringing this perspective to our attention.  I feel responsible for the lack of clarity here.  

When Tina started out, it was just the UI part which is licensed under Apache 2.0.  Hence the name, TinaCMS = **T**ina **i**s **n**ot **a** CMS (meaning, it wasn't a "full" CMS).  Back then, Tina could communicate with any API (like Shopify or another headless CMS) but most users thought it didn't go far enough and wanted a more end-to-end solution. They wanted a _complete_ CMS. So we built Tina Cloud, a hosted (closed-source) headless backend.  So we were left with an open-source CMS Admin UI and a proprietary headless backend. 

We then built a [self-hosted alternative](https://tina.io/docs/self-hosted/overview/) to Tina Cloud, but licensed it under a fair-source license which limits usage for larger projects (usually by larger corporations).  We took the stance that if a large corporation wanted to use the self-hosted option, they should enter into a custom agreement with us since we're a small team and need to pay software engineers to build and maintain that solution. 

We tried to make the open-source and non-open-source distinction clear in the [self-hosted docs](https://tina.io/docs/self-hosted/overview/) and in the [Tina License](https://github.com/tinacms/tinacms/blob/main/LICENSE), but obviously didn't go far enough.  

I'm sorry for the confusion.  I'll make sure to discuss your points with the team today.  If you want to chat more about this, feel free to DM me in [Discord](https://discord.com/invite/zumN63Ybpf) (username `ScottG`).

**Reply by Hubro** [reply]
Thanks for clarifying. I completely understand and agree that corporations shouldn't use software for free and contribute nothing back to the authors. However, unless I've misunderstood something, I would argue that the current "free tier" of Tina is so restrictive that it can only be used for demos and testing.

As soon as you add a blog, news feed, gallery or anything similar, it's only matter of time before even the smallest production website exceeds 500 documents. Many (maybe most?) websites will have dynamic content like that.

If your goal really is that individuals, startups and small companies should be able to use Tina for free, it might be a good idea to tweak the license.

Why not go with a license that says the software can be used for free, but can't be used to create a competing product to Tina Cloud? Plus, provide premium support for enterprise. That feels a lot more in the spirit of open source, smaller users could actually embrace your product and deploy it without constantly looking out for the license limits.

**Reply by scottgallant** [reply]
> If your goal really is that individuals, startups and small companies should be able to use Tina for free, it might be a good idea to tweak the license.

👍 We'll discuss increasing the document limit. @Hubro, do you suspect you'll hit the 500-document limit for your project? If so, feel free to reach out to me in Discord and I'll see what we can do for the time being. 

> Why not go with a license that says the software can be used for free, but can't be used to create a competing product to Tina Cloud? Plus, provide premium support for enterprise. That feels a lot more in the spirit of open source, smaller users could actually embrace your product and deploy it without constantly looking out for the license limits.

After speaking with a few large companies, we learned that they would often just choose the open-source solution.  In the future we might consider going fully open-source (or open-core), especially if we develop proprietary features that could really improve the experience for large orgs and give them a reason to become paying customers.

**Reply by Hubro** [reply]
> do you suspect you'll hit the 500-document limit for your project?

I think all my projects will hit this limit, as long as they include any kind of blog, news feed or similar. It's really just a matter of time, isn't it?

Another alternative could be to remove the document limit. I feel like user count and role count is correlated to company size, but document count will increase over time for most websites.

**Reply by scottgallant** [reply]
@Hubro, it would be helpful for us to understand what kind of projects you would be working on so we can better understand where you're coming from. Are these marketing sites for small businesses, internal sites at a larger company, personal sites, etc?  

Also, would it be helpful if we published pricing for the self-hosted so you knew  you weren't going to hit "contact us" wall once you reached any limit?   Here's what we're thinking that might look like: 

<img width="593" alt="Screenshot 2023-05-26 at 3 46 17 PM" src="https://github.com/tinacms/tinacms/assets/776019/42f38cf0-b96e-4a87-b6e3-1e3b19a3127a">

**Reply by Hubro** [reply]
@scottgallant As an example, my most recent project was a new website for a recurring music festival. This website has a news section, as well as a list of events, sponsors and artists.

This festival is largely supported by volunteers and they don't have much of a budget for their website. Their website hosting bill is around 10$ per month, plus however much they pay SendGrid for their newsletter. They host a large festival annually and several small events throughout the year. I estimate they would break the 500 document limit in around 2 years.

I would _never_ consider using Tina for this project with the current license. For a website with a less active news section it might take longer to hit the document limit, but I would still never consider delivering a website using Tina and impose on the customer to monitor their document count, and a time limit before their hosting bill increases 5-10 fold.
